### ISSUE TITLE
defaulttags seems weird

### ISSUE THREAD

**Issue by kbilsted** [mvno/Okanshi#89]
Adding default tags seems a bit cumbersome from C# look at this example, of adding 2 tags to the default tags.

			var duplicateTag = new Tag("a", "b");

			OkanshiMonitor.DefaultTags = 
				OkanshiMonitor.DefaultTags
				.Union(new[] {duplicateTag, duplicateTag})
				.ToArray();

why are we not exposing `DefaultTags` as a `HashSet` rather than a `seq` ?

To achieve code like 
    
    OkanshiMonitor.DefaultTags.Add(duplicateTag);
    OkanshiMonitor.DefaultTags.Add(duplicateTag);

**Comment by kichristensen** [comment]
I can't remember for sure why, it has done like this. But I would guess it is to indicate that DefaultTags shouldn't be mutated, only be set once to avoid changing default tags after initial configuration.

Also I think you made it more difficult than needed, you could easily do something like this:

```
var tag = new Tag("a", "b");
OkanshiMonitor.DefaultTags = new[] { tag, tag };
```

**Comment by kbilsted** [comment]
your example will remove all existing tags, mine retain existing..

**Comment by kichristensen** [comment]
That is done in the setter anyway, the content of the default tags are not mutable

**Comment by kbilsted** [comment]
the following code fails

         	[Fact]
		public void When_adding_tags_multiple_times_Then_all_tags_are_added()
		{
			var tag1 = new Tag("a", "b");
			var tag2 = new Tag("c", "d");

			OkanshiMonitor.DefaultTags = new[] { tag1 };
			OkanshiMonitor.DefaultTags = new[] { tag2 };

			OkanshiMonitor.DefaultTags.Should().BeEquivalentTo(new[] { tag1, tag2 });
		}

**Comment by kichristensen** [comment]
Yes, as expected, what it your point? That it is unexpected or?

**Comment by kichristensen** [comment]
Ahh my previous answer was not clear. The setters always clears the HashSet, currently by design

**Comment by kichristensen** [comment]
It think main "confusion" here is that the static class is used both for configuration and usage, perhaps we should rethink that

**Comment by kbilsted** [comment]
To be honest I don't think the builder pattern looks good as an example code but won't change production code neither for good or worse. Fact is configuration is a miniscule part of an application. 

I would just avoid trying to protect the user when there is no protection. And just change the type to a hashsset. Of course it would be fun to code a builder so that's an option too. My PR would be just a hashset though ;)

**Comment by kichristensen** [comment]
I'm not sure talking about a specific solution here, it could be a builder, it could be some else. I think think it is clearer that it isn't recommended to modified the default tags, to pass the full list instead of allowing mutation, but it could just be me.

**Comment by kbilsted** [comment]
Well id prefer we not make too many assumptions as how the user will or want to use the api. Thus it should be easy to administrate . Exposing a hashset makes this easy and automatically tells the user how duplicate tags are handled

**Comment by kichristensen** [comment]
closed in #92
